runtime/j9vm/java9vmi.c

```diff
@@ -381,11 +381,11 @@ trcModulesCreationPackage(J9VMThread * currentThread, J9Module * fromModule, j9o
 	J9InternalVMFunctions const * const vmFuncs = currentThread->javaVM->internalVMFunctions;
 	char moduleNameBuf[J9VM_PACKAGE_NAME_BUFFER_LENGTH];
 	char *moduleNameUTF = vmFuncs->copyStringToUTF8WithMemAlloc(
-		currentThread, fromModule->moduleName, J9_STR_NONE, "", moduleNameBuf, J9VM_PACKAGE_NAME_BUFFER_LENGTH);
+		currentThread, fromModule->moduleName, J9_STR_NONE, "", 0, moduleNameBuf, J9VM_PACKAGE_NAME_BUFFER_LENGTH, NULL);
 #if J9VM_JAVA9_BUILD < 156
 	char packageNameBuf[J9VM_PACKAGE_NAME_BUFFER_LENGTH];
 	char *packageNameUTF = vmFuncs->copyStringToUTF8WithMemAlloc(
```

Comment:
`null` (native OOM) check for `packageNameUTF` was missing as well. There are quite a few similar cases in this file as well.
How about let API itself set `setNativeOutOfMemoryError` when `j9mem_allocate_memory` failed?
P.S. rebasing seems bring up more questions :)

Reply:
@JasonFengJ9 this seems like pre-existing bugs in the code. I think we are starting to scope creep in this PR. This PR is targeted towards performance improvements of String related APIs. May I suggest we focus only on those? We can open up issues for related code in the area of these changes which need additional fixes. Does that sound reasonable?

I'm afraid there will be more and more places to fix pre-existing functional issues not related to this _performance_ change. I suggest we leave these for a subsequent PR.